pulsar-broker/src/test/java/org/apache/pulsar/broker/intercept/CounterBrokerInterceptor.java

```diff
@@ -163,6 +170,13 @@ public void messageProduced(ServerCnx cnx, Producer producer, long startTimeNs,
                     producer.getTopic().getName(), producer.getProducerName());
         }
         messageCount.incrementAndGet();
+
+        var topicName = TopicName.get(producer.getTopic().getName());
+        var builder = Attributes.builder()
+                .put(OpenTelemetryAttributes.PULSAR_TOPIC, topicName.getPartitionedTopicName());
+        var attributes = builder.build();
+
+        messageCounter.add(1, attributes);
```

Comment:
Since many will record topic-level metrics, we need a common infra to allow you get an AttributesBuilder already filled with the topic level attributes.
@dragosvictor is currently working on implementing many OTel metrics, I suggest you get his plan for this, so you won't overlap.

Reply:
This is just a demo, it will actually be a custom attribute.